test/dynamo/test_autograd_function.py

```diff
@@ -0,0 +1,115 @@
+# Owner(s): ["module: dynamo"]
+
+import math
+
+import torch
+
+import torch._dynamo.test_case
+import torch._dynamo.testing
+
+
+class CustomFunc1(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, foo):
+        return foo + foo
+
+    @staticmethod
+    def backward(ctx, grad_output):
+        return grad_output
+
```

Comment:
So there is a second type of autograd.Function that has a `setup_context` staticmethod where the forward doesn't actually accept a ctx object (https://pytorch.org/docs/stable/notes/extending.html#example). We don't need to support it in this PR, but it would be good to test that that does not completely blow up underneath torch.compile (today I would expect it to graph break)

Reply:
I can add a test for this :) It should gracefully graph break here.

Reply:
Test added.